src/utils/api/getBlocks.ts

```diff
@@ -0,0 +1,41 @@
+//
+// API call helper for getBlocks, not meant to be called directly (meant to be used inside the APIProvider)
+//
+
+import { API, IAPICall } from "./api";
+import { BASEURL } from "./constants";
+
+/* &requestType=getBlocks
+&firstIndex=1000
+&lastIndex=1001 */
+
+interface IGetBlocksPayload extends IAPICall {
+  data: {
+    firstIndex: number;
+    lastIndex: number;
+  };
+}
+
+async function getBlocks(startBlock: number, endBlock: number) {
+  let result;
+
+  const options: IGetBlocksPayload = {
+    url: BASEURL,
+    method: "POST",
+    requestType: "getBlocks",
+    data: {
+      firstIndex: startBlock,
```

Comment:
any logic to ensure that the start and end blocks are in the right order? meaning start block is < end block? can it be <= ?

Reply:
I added some logic which does the trick for now, but eventually I will integrate this return with a broader set of improvements regarding error handling (setting up a constants file with various error text, supplying it back to the user via notification system or console.error depending on how far up it's desired to bubble the error).